backend/python-functions/data_ingest.py

```diff
@@ -0,0 +1,147 @@
+import csv
+from google.cloud import firestore
+from google.cloud import storage
+
+REFERENCE_FACILITIES_COLLECTION_ID = 'reference_facilities'
+
+def build_covid_case_counts(row):
+    covid_case_counts = {
+        'popDeaths': row['Pop Deaths'],
+        'popTested': row['Pop Tested'],
+        'popTestedNegative': row['Pop Tested Negative'],
+        'popTestedPositive': row['Pop Tested Positive'],
+        'staffDeaths': row['Staff Deaths'],
+        'staffTested': row['Staff Tested'],
+        'staffTestedNegative': row['Staff Tested Negative'],
+        'staffTestedPositive': row['Staff Tested Positive'],        
+    }
+    
+    # Remove empty strings and convert non-empty strings to integers
+    return {k: int(v) for k, v in covid_case_counts.items() if v}
+
+def download_covid_case_data(bucket_name, file_name):
+    storage_client = storage.Client()
+    blob = storage_client.get_bucket(bucket_name).get_blob(file_name) 
+
+    download_location = f'/tmp/{file_name}'
+
+    blob.download_to_filename(download_location)    
+
+    return download_location
+
+def reshape_facilities_data(file_location):
+    """
+    This function reshapes the daily Covid case data provided in CSV format into a 
+    dictionary structure that: 1) Removes repetitive and unwanted data and 2) Aligns
+    with how we ultimately persist our documents within Firestore.
+
+    For example, the following CSV:
+
+    Date,Facility Type,State,Canonical Facility Name,Pop Tested,Pop Tested Positive, etc.
+    2020-06-01,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+    2020-06-02,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+    2020-06-03,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+    2020-06-04,State Prisons,Colorado,Denver Reception & Diagnostic Center,253,2,,0,,,,,,,
+
+    Would be reshaped to:
+
+    {
+      "Colorado::Denver Reception & Diagnostic Center": {
+        "canonicalName": "Denver Reception & Diagnostic Center",
+        "facilityType": "State Prisons",
+        "state": "Colorado",
```

Comment:
some minor housekeeping but could we call this field `stateName` instead? trying to standardize references to US states versus application state per #584

Reply:
Will do.  Are we also intending to update the document schema for the reference facilities in Firestore to use `stateName` instead of `state`?

Reply:
Sorry, let me re-phrase.  If we make this change, this will set the document field to `stateName` in the `reference_facilities` collection's documents.  Is that cool?

Reply:
we will have to make some minor code changes probably but yes

Reply:
Got it.  Will make the update